crates/runtime/src/starknet/context.rs

```diff
@@ -130,39 +86,112 @@ pub fn build_transaction_context() -> AccountTransactionContext {
+        tx_info: build_tx_info(),
+    }
 }
 
-pub fn set_max_steps(entry_point_ctx: &mut EntryPointExecutionContext, max_n_steps: u32) {
-    entry_point_ctx.block_context.invoke_tx_max_n_steps = max_n_steps;
+#[must_use]
+pub fn build_context(block_info: &BlockInfo) -> EntryPointExecutionContext {
+    let transaction_context = Arc::new(build_transaction_context(block_info));
 
+    EntryPointExecutionContext::new(transaction_context, ExecutionMode::Execute, false).unwrap()
+}
+
+pub fn set_max_steps(entry_point_ctx: &mut EntryPointExecutionContext, max_n_steps: u32) {
     // override it to omit [`EntryPointExecutionContext::max_steps`] restrictions
     entry_point_ctx.vm_run_resources = RunResources::new(max_n_steps as usize);
 }
 
-#[derive(Copy, Clone, Serialize, Deserialize, Debug)]
-pub struct BlockInfo {
+// We need to be copying those 1:1 for serialization (caching purposes)
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct SerializableBlockInfo {
     pub block_number: BlockNumber,
-    pub timestamp: BlockTimestamp,
+    pub block_timestamp: BlockTimestamp,
     pub sequencer_address: ContractAddress,
+    pub gas_prices: SerializableGasPrices,
+    // A field which indicates if EIP-4844 blobs are used for publishing state diffs to l1
+    // This has influence on the cost of publishing the data on l1
+    pub use_kzg_da: bool,
+}
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct SerializableGasPrices {
+    eth_l1_gas_price: NonZeroU128,
+    strk_l1_gas_price: NonZeroU128,
+    eth_l1_data_gas_price: NonZeroU128,
+    strk_l1_data_gas_price: NonZeroU128,
+}
+impl Default for SerializableGasPrices {
+    fn default() -> Self {
+        Self {
+            eth_l1_gas_price: NonZeroU128::try_from(100 * u128::pow(10, 9)).unwrap(),
+            strk_l1_gas_price: NonZeroU128::try_from(100 * u128::pow(10, 9)).unwrap(),
+            eth_l1_data_gas_price: NonZeroU128::try_from(u128::pow(10, 6)).unwrap(),
+            strk_l1_data_gas_price: NonZeroU128::try_from(u128::pow(10, 9)).unwrap(),
+        }
+    }
 }
 
-impl Default for BlockInfo {
+impl Default for SerializableBlockInfo {
     fn default() -> Self {
         Self {
             block_number: BlockNumber(DEFAULT_BLOCK_NUMBER),
-            timestamp: BlockTimestamp::default(),
-            sequencer_address: contract_address!(SEQUENCER_ADDRESS),
+            block_timestamp: BlockTimestamp::default(),
+            sequencer_address: ContractAddress(patricia_key!(SEQUENCER_ADDRESS)),
+            gas_prices: SerializableGasPrices {
+                eth_l1_gas_price: NonZeroU128::try_from(100 * u128::pow(10, 9)).unwrap(),
+                strk_l1_gas_price: NonZeroU128::try_from(100 * u128::pow(10, 9)).unwrap(),
+                eth_l1_data_gas_price: NonZeroU128::try_from(u128::pow(10, 6)).unwrap(),
+                strk_l1_data_gas_price: NonZeroU128::try_from(u128::pow(10, 9)).unwrap(),
+            },
```

Comment:
`SerializableGasPrices::default()` perhaps?

Reply:
Ah, yes. that was the purpose of the above 😅